Consul/Client_GetRequests.cs

```diff
@@ -39,9 +39,19 @@ public class GetRequest<TOut> : ConsulRequest
     {
         public QueryOptions Options { get; set; }
 
-        public IEncodable Filter { get; set; }
+        public IEncodable Filter { get; }
```

Comment:
I think that public setter is not needed here and it is better to aim for immutability so I would suggest removing it. `Options` field also should not have public setter but now "too late" to remove it as it would be binary incompatible change.

Reply:
It could still be removed in the next major version though. Let's make a note?

Reply:
I tend to agree with @mfkl Removing a setter sounds like one for a major version. Marking it deprecated might be a good idea though?

Reply:
Unfortunately it is not possible to obsolete individual accessors in a property (https://stackoverflow.com/questions/4108300/is-it-possible-to-use-obsolete-attribute-on-only-a-getter-or-a-setter-of-a-prope/55508328), I'll make a comment to remove the setter in next major release.